### PR TITLE
#453 Moved cli code into separate module

### PR DESCRIPTION
**Pull request by frankscholten**

### CODE REVIEW

**Comment by containersol** [comment]
SonarQube analysis reported 16 issues:
* CRITICAL 2 critical
* MAJOR 10 major
* MINOR 4 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MINOR [Remove the declaration of thrown exception 'com.containersol.minimesos.MinimesosException' which is a runtime exception.](https://github.com/ContainerSolutions/minimesos/blob/cc83a10666761c1f9169fa445e0d1020f4d307c2/cli/src/main/java/com/containersol/minimesos/main/Command.java#L22) [rule](https://ci.infra.container-solutions.com/sonar/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
* MAJOR [Replace this usage of System.out or System.err by a logger.](https://github.com/ContainerSolutions/minimesos/blob/cc83a10666761c1f9169fa445e0d1020f4d307c2/cli/src/main/java/com/containersol/minimesos/main/CommandInfo.java#L23) [rule](https://ci.infra.container-solutions.com/sonar/coding_rules#rule_key=squid%3AS106)
* MAJOR [Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation.](https://github.com/ContainerSolutions/minimesos/blob/cc83a10666761c1f9169fa445e0d1020f4d307c2/cli/src/main/java/com/containersol/minimesos/main/CommandInfo.java#L27) [rule](https://ci.infra.container-solutions.com/sonar/coding_rules#rule_key=squid%3AS1186)
* MINOR [Remove the declaration of thrown exception 'com.containersol.minimesos.MinimesosException' which is a runtime exception.](https://github.com/ContainerSolutions/minimesos/blob/cc83a10666761c1f9169fa445e0d1020f4d307c2/cli/src/main/java/com/containersol/minimesos/main/CommandInit.java#L52) [rule](https://ci.infra.container-solutions.com/sonar/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
* CRITICAL [Either log or rethrow this exception.](https://github.com/ContainerSolutions/minimesos/blob/cc83a10666761c1f9169fa445e0d1020f4d307c2/cli/src/main/java/com/containersol/minimesos/main/CommandInit.java#L74) [rule](https://ci.infra.container-solutions.com/sonar/coding_rules#rule_key=squid%3AS1166)
* MINOR [Remove the declaration of thrown exception 'com.containersol.minimesos.MinimesosException' which is a runtime exception.](https://github.com/ContainerSolutions/minimesos/blob/cc83a10666761c1f9169fa445e0d1020f4d307c2/cli/src/main/java/com/containersol/minimesos/main/CommandInstall.java#L68) [rule](https://ci.infra.container-solutions.com/sonar/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
* MAJOR [Remove this empty class, write its code or make it an "interface".](https://github.com/ContainerSolutions/minimesos/blob/cc83a10666761c1f9169fa445e0d1020f4d307c2/cli/src/main/java/com/containersol/minimesos/main/CommandMain.java#L6) [rule](https://ci.infra.container-solutions.com/sonar/coding_rules#rule_key=squid%3AS2094)
* MAJOR [Replace this usage of System.out or System.err by a logger.](https://github.com/ContainerSolutions/minimesos/blob/cc83a10666761c1f9169fa445e0d1020f4d307c2/cli/src/main/java/com/containersol/minimesos/main/CommandState.java#L22) [rule](https://ci.infra.container-solutions.com/sonar/coding_rules#rule_key=squid%3AS106)
* MAJOR [Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation.](https://github.com/ContainerSolutions/minimesos/blob/cc83a10666761c1f9169fa445e0d1020f4d307c2/cli/src/main/java/com/containersol/minimesos/main/CommandState.java#L26) [rule](https://ci.infra.container-solutions.com/sonar/coding_rules#rule_key=squid%3AS1186)
* MAJOR [Replace this usage of System.out or System.err by a logger.](https://github.com/ContainerSolutions/minimesos/blob/cc83a10666761c1f9169fa445e0d1020f4d307c2/cli/src/main/java/com/containersol/minimesos/main/CommandUp.java#L49) [rule](https://ci.infra.container-solutions.com/sonar/coding_rules#rule_key=squid%3AS106)
* ... 6 more

**Comment by containersol** [comment]
SonarQube analysis reported 12 issues:
* CRITICAL 2 critical
* MAJOR 6 major
* MINOR 4 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MINOR [Remove the declaration of thrown exception 'com.containersol.minimesos.MinimesosException' which is a runtime exception.](https://github.com/ContainerSolutions/minimesos/blob/a342e6fd87e769d7729b4dc7f08cded1f8320ca0/cli/src/main/java/com/containersol/minimesos/main/Command.java#L22) [rule](https://ci.infra.container-solutions.com/sonar/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
* MAJOR [Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation.](https://github.com/ContainerSolutions/minimesos/blob/a342e6fd87e769d7729b4dc7f08cded1f8320ca0/cli/src/main/java/com/containersol/minimesos/main/CommandInfo.java#L27) [rule](https://ci.infra.container-solutions.com/sonar/coding_rules#rule_key=squid%3AS1186)
* MINOR [Remove the declaration of thrown exception 'com.containersol.minimesos.MinimesosException' which is a runtime exception.](https://github.com/ContainerSolutions/minimesos/blob/a342e6fd87e769d7729b4dc7f08cded1f8320ca0/cli/src/main/java/com/containersol/minimesos/main/CommandInit.java#L52) [rule](https://ci.infra.container-solutions.com/sonar/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
* CRITICAL [Either log or rethrow this exception.](https://github.com/ContainerSolutions/minimesos/blob/a342e6fd87e769d7729b4dc7f08cded1f8320ca0/cli/src/main/java/com/containersol/minimesos/main/CommandInit.java#L74) [rule](https://ci.infra.container-solutions.com/sonar/coding_rules#rule_key=squid%3AS1166)
* MINOR [Remove the declaration of thrown exception 'com.containersol.minimesos.MinimesosException' which is a runtime exception.](https://github.com/ContainerSolutions/minimesos/blob/a342e6fd87e769d7729b4dc7f08cded1f8320ca0/cli/src/main/java/com/containersol/minimesos/main/CommandInstall.java#L68) [rule](https://ci.infra.container-solutions.com/sonar/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
* MAJOR [Remove this empty class, write its code or make it an "interface".](https://github.com/ContainerSolutions/minimesos/blob/a342e6fd87e769d7729b4dc7f08cded1f8320ca0/cli/src/main/java/com/containersol/minimesos/main/CommandMain.java#L6) [rule](https://ci.infra.container-solutions.com/sonar/coding_rules#rule_key=squid%3AS2094)
* MAJOR [Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation.](https://github.com/ContainerSolutions/minimesos/blob/a342e6fd87e769d7729b4dc7f08cded1f8320ca0/cli/src/main/java/com/containersol/minimesos/main/CommandState.java#L26) [rule](https://ci.infra.container-solutions.com/sonar/coding_rules#rule_key=squid%3AS1186)
* MAJOR [Rename "numAgents" which hides the field declared at line 42.](https://github.com/ContainerSolutions/minimesos/blob/a342e6fd87e769d7729b4dc7f08cded1f8320ca0/cli/src/main/java/com/containersol/minimesos/main/CommandUp.java#L78) [rule](https://ci.infra.container-solutions.com/sonar/coding_rules#rule_key=squid%3AHiddenFieldCheck)
* MAJOR [Use isEmpty() to check whether the collection is empty or not.](https://github.com/ContainerSolutions/minimesos/blob/a342e6fd87e769d7729b4dc7f08cded1f8320ca0/cli/src/main/java/com/containersol/minimesos/main/CommandUp.java#L83) [rule](https://ci.infra.container-solutions.com/sonar/coding_rules#rule_key=squid%3AS1155)
* MINOR [Move this constructor to comply with Java Code Conventions.](https://github.com/ContainerSolutions/minimesos/blob/a342e6fd87e769d7729b4dc7f08cded1f8320ca0/cli/src/main/java/com/containersol/minimesos/main/Main.java#L72) [rule](https://ci.infra.container-solutions.com/sonar/coding_rules#rule_key=squid%3AS1213)
* ... 2 more

**Comment by containersol** [comment]
SonarQube analysis reported 3 issues:
* CRITICAL 1 critical
* MAJOR 1 major
* MINOR 1 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MINOR [Remove the declaration of thrown exception 'com.containersol.minimesos.MinimesosException' which is a runtime exception.](https://github.com/ContainerSolutions/minimesos/blob/d73cb152208199675ab1cdd153e22847660c77fa/cli/src/main/java/com/containersol/minimesos/main/CommandInstall.java#L68) [rule](https://ci.infra.container-solutions.com/sonar/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
* MAJOR [The Cyclomatic Complexity of this method "run" is 16 which is greater than 10 authorized.](https://github.com/ContainerSolutions/minimesos/blob/d73cb152208199675ab1cdd153e22847660c77fa/cli/src/main/java/com/containersol/minimesos/main/Main.java#L82) [rule](https://ci.infra.container-solutions.com/sonar/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
* CRITICAL [Either log or rethrow this exception.](https://github.com/ContainerSolutions/minimesos/blob/d73cb152208199675ab1cdd153e22847660c77fa/cli/src/main/java/com/containersol/minimesos/main/Main.java#L90) [rule](https://ci.infra.container-solutions.com/sonar/coding_rules#rule_key=squid%3AS1166)

File: cli/src/main/java/com/containersol/minimesos/main/Main.java [hunk 1]
```diff
@@ -67,69 +69,87 @@ public static void main(String[] args) {
...
+                if (parsedCommand.validateParameters()) {
+                    parsedCommand.execute();
+                } else {
+                    printUsage(jc.getParsedCommand());
+                    return EXIT_CODE_ERR;
+                }
+            }
+
+            return EXIT_CODE_OK;
+        } catch (Exception ex) {
```
Comment (containersol) [inline cli/src/main/java/com/containersol/minimesos/main/Main.java]: CRITICAL Either log or rethrow this exception. [rule](https://ci.infra.container-solutions.com/sonar/coding_rules#rule_key=squid%3AS1166)

File: cli/src/main/java/com/containersol/minimesos/main/Main.java [hunk 2]
```diff
@@ -67,69 +69,87 @@ public static void main(String[] args) {
...
-    public Main() {
-        jc = new JCommander(this);
-        jc.setProgramName("minimesos");
-    }
 
     public void setOutput(PrintStream output) {
         this.output = output;
     }
 
     public int run(String[] args) {
```
Comment (containersol) [inline cli/src/main/java/com/containersol/minimesos/main/Main.java]: MAJOR The Cyclomatic Complexity of this method "run" is 14 which is greater than 10 authorized. [rule](https://ci.infra.container-solutions.com/sonar/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)

**Comment by containersol** [comment]
SonarQube analysis reported 2 issues:
* CRITICAL 1 critical
* MAJOR 1 major

Watch the comments in this conversation to review them.

**Comment by frankscholten** [comment]
LGTM

**Comment by containersol** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.

**Pull request merged**